### DISCUSSION TITLE
Error creating a user

### DISCUSSION THREAD

**Discussion by richardltc** [pocketbase/pocketbase#1655] [Q&A] [answered]
Hi,

I'm trying to add a user using SvelteKit, and am getting the following error:
```
ClientResponseError 400: Failed to create record.
    at new ClientResponseError (file:///home/richard/projects/richardmace.co.uk/pocketbase-sveltekit-auth/node_modules/pocketbase/dist/pocketbase.es.mjs:1:2285)
    at file:///home/richard/projects/richardmace.co.uk/pocketbase-sveltekit-auth/node_modules/pocketbase/dist/pocketbase.es.mjs:1:40244
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async default (/src/routes/register/+page.server.ts:10:7)
    at async Module.handle_action_request (/node_modules/@sveltejs/kit/src/runtime/server/page/actions.js:154:16)
    at async Module.render_page (/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:66:20)
    at async resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:365:17)
    at async Object.handle (/src/hooks.server.ts:16:20)
    at async Module.respond (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:236:20)
    at async file:///home/richard/projects/richardmace.co.uk/pocketbase-sveltekit-auth/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:467:22 {
  url: 'http://127.0.0.1:8090/api/collections/users/records',
  status: 400,
  data: {
    code: 400,
    message: 'Failed to create record.',
    data: { password: [Object] }
  },
  isAbort: false,
  originalError: null
}
```
Here is the code that causes the issue:

```
    default: async ({locals, request}) => {
        const data = Object.fromEntries(await request.formData()) as {
            email: string;
            password: string;
            passwordConfirm: string;

        };
```

However, when I use this code instead, it does work:

```
        const data = {
             // "username": "test_username2",
             "email": "test@example2.com",
             // "emailVisibility": true,
             "password": "12345678",
             "passwordConfirm": "12345678",
             // "name": "test2"
         };
```

Any ideas of what I'm doing wrong?

Thanks in advance

Richard

**Comment by ganigeorgiev** [comment]
The error response indicates that something is wrong with the password field.
You can `try/catch` your call and inspect the HTTP error response data details using `err.data.data` (_this will be soft-deprecated in the future in favour of `err.response.data`_). For list with all available fields you can check https://github.com/pocketbase/js-sdk#error-handling.

Alternatively, you can also start PocketBase with `--debug` and/or check the request logs from the "Admin UI > Logs".

**Reply by ganigeorgiev** [reply]
Also make sure that `await request.formData()` resolves into the expected value, because I think you may have to chain `.entries()` so that you can create an object from it with the `Object.fromEntries()`, if that's what you want, something like (_I haven't tested it and I'm not sure whether sveltekit applies any special transformations to the FormData object_):

```js
Object.fromEntries((await request.formData()).entries());
```

**Comment by richardltc** [comment, marked as answer]
Thank you. I managed to find the logs, and I could see that the password length was not long enough :) Thanks again
